lib/Sema/DerivedConformanceCodable.cpp

```diff
@@ -314,15 +357,11 @@ static bool validateCodingKeysEnum(const DerivedConformance &derived,
   // If there are any vars left in the type which don't have a default value
   // (for Decodable), then this decl doesn't match.
   bool varDeclsAreValid = true;
+  auto extraKeys = llvm::SmallVector<EnumElementDecl *, 4>();
```

Comment:
Nit: We generally prefer to let the default constructor take care of values like this;
```suggestion
llvm::SmallVector<EnumElementDecl *, 4> extraKeys;
```